### PR TITLE
AC-376:Allow users to show log file in the app

### PR DESCRIPTION
**Pull request by DefCon-007**
screenshot_20170218-200716 1

### CODE REVIEW

**Comment by AvijitGhosh82** [comment]
Neat! Thanks for the PR @DefCon-007. Merging.

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/settings/SettingsActivity.java [hunk 1]
```diff
@@ -46,7 +49,7 @@ protected void onCreate(Bundle savedInstanceState) {
         }
 
         // Create the presenter
-        new SettingsPresenter(settingsFragment, mOpenMRSLogger);
+        new SettingsPresenter(settingsFragment, mOpenMRSLogger,logsTV);
```
Comment (AdamGrzybkowski) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/settings/SettingsActivity.java]: @DefCon-007  @AvijitGhosh82 We shouldn't pass Textview to the presenter

File: openmrs-client/src/main/java/org/openmrs/mobile/activities/settings/SettingsFragment.java [hunk 2]
```diff
@@ -57,7 +64,25 @@ public void onPause() {
...
+        String aBuffer = "";
+        try {
+                File myFile = new File(logFilename);
+                FileInputStream fIn = new FileInputStream(myFile);
+                BufferedReader myReader = new BufferedReader(new InputStreamReader(fIn));
+                String aDataRow;
+                while ((aDataRow = myReader.readLine()) != null) {
+                        aBuffer += aDataRow;
+                    }
+                myReader.close();
```
Comment (AdamGrzybkowski) [inline openmrs-client/src/main/java/org/openmrs/mobile/activities/settings/SettingsFragment.java]: @DefCon-007 This logic should be handled by the presenter

**Comment by DefCon-007** [comment]
Ok I will make a new activity for the logs.

**Comment by tmarzeion** [comment]
@AvijitGhosh82 
We are using MVP here and its really bad practice to pass View objects to Presenter.
Please read why we did this:
https://github.com/googlesamples/android-architecture
I created ticket and assign it to You, please fix that:
https://issues.openmrs.org/browse/AC-388

**Comment by AvijitGhosh82** [comment]
@tmarzeion This is fixed in #360 . Closing issue.

**Pull request merged**